advisor/src/main/kotlin/advisors/GitHubDefects.kt

```diff
@@ -89,8 +107,18 @@ class GitHubDefects(name: String, config: GitHubDefectsConfiguration) : AdvicePr
         const val DEFAULT_PARALLEL_REQUESTS = 4
     }
 
-    class Factory : AbstractAdviceProviderFactory<GitHubDefects>("GitHubDefects") {
-        override fun create(config: AdvisorConfiguration) = GitHubDefects(type, config.forProvider { gitHubDefects })
+    class Factory : AdviceProviderFactory<GitHubDefectsConfiguration>("GitHubDefects") {
+        override fun create(config: GitHubDefectsConfiguration) = GitHubDefects(type, config)
+
+        override fun parseConfig(options: Options, secrets: Options) =
+            GitHubDefectsConfiguration(
+                token = secrets["token"],
+                endpointUrl = options["endpointUrl"],
+                labelFilter = options["labelFilter"]?.split(",")?.map { it.trim() }
+                    ?: listOf("!duplicate", "!enhancement", "!invalid", "!question", "*"),
```

Comment:
This now duplicates the default value of `GitHubDefectsConfiguration.labelFilter`, which IMO is not so nice. In [my old draft](https://github.com/oss-review-toolkit/ort/pull/6613) I tried to [solve this by extracting default values to constants](https://github.com/oss-review-toolkit/ort/pull/6613/commits/0db7e0f29d750b6e336cff02658f1673b1e8673d#diff-c683de376a76556ce1a5c1d790017956b5726c1b270ca08a1e3e2d61a76054cdR106-R112) (not fully implemented).

This raises the general question where default values should be applied. I see basically tree options:

1. in `parseConfig()`,
2. in the constructor of the output config class,
3. in the consumer of the config class.

My current preference would be 1., and I'd propose a clean-up commit that removes the default parameters from `GitHubDefectsConfiguration`'s constructor.

Reply:
Would you be fine with doing that clean up commit outside of this PR?

Reply:
Yes, if it's resolved within reasonable time, before we forget about it.